components/FlexibleHeader/tests/unit/FlexibleHeaderShiftedOffscreenWithShiftBehaviorHideableTests.m

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialFlexibleHeader.h"
+
+// Validates the behavior of isShiftedOffscreen and flexible header shifting when shiftBehavior
+// is MDCFlexibleHeaderShiftBehaviorHideable. In particular, it demonstrates that calling
+// shiftHeaderOffScreenAnimated: causes the header to be permanently shifted off-screen; i.e. the
+// user cannot drag the header back on-screen.
+@interface FlexibleHeaderShiftedOffscreenWithShiftBehaviorHideableTests : XCTestCase
+@property(nonatomic, strong) MDCFlexibleHeaderView *fhvc;
+@property(nonatomic, strong) UIScrollView *scrollView;
+@end
+
+@implementation FlexibleHeaderShiftedOffscreenWithShiftBehaviorHideableTests
```

Comment:
do we want to add tests for setting shiftBehavior = MDCFlexibleHeaderShiftBehaviorHideable and making sure the outcome is correct?

Reply:
All of these tests validate the behavior for when `shiftBehavior = MDCFlexibleHeaderShiftBehaviorHideable`; see line 34 below.